src/wasi_unstable/mod.rs

```diff
@@ -49,9 +50,25 @@ pub type FileStat = __wasi_filestat_t;
 pub type CIoVec = __wasi_ciovec_t;
 pub type IoVec = __wasi_iovec_t;
 pub type Subscription = __wasi_subscription_t;
-pub type Event = __wasi_event_t;
 pub type Prestat = __wasi_prestat_t;
 
+// should have exactly the same layout as __wasi_event_t
+// TODO: add a more thorough test
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct Event {
+    pub userdata: __wasi_userdata_t,
```

Comment:
We might as well use the Rust-style type names, like `Userdata`, and so on for the other fields.